pkg/builder/controller.go

```diff
@@ -289,6 +290,9 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	if err != nil {
 		return err
 	}
+	if gvk.Kind == "" {
```

Comment:
Is this possible? GVK should always include a Kind or not?

Reply:
Exactly. I only added this to be 100% safe against panics in l.321 even if code elsewhere changes.

But I'm fine with dropping it if you prefer.

Reply:
I've dropped it. We're checking that object is not nil one layer above + getGvk always returns either a gvk or an error. So should be safe enough.